pkg/cmd/checkendpoints/start_stop_controllers.go

```diff
@@ -0,0 +1,90 @@
+package checkendpoints
+
+import (
+	"context"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsinformersv1 "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions/apiextensions/v1"
+	apiextensionslistersv1 "k8s.io/apiextensions-apiserver/pkg/client/listers/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+)
+
+// timeToStartController will either close its ready chan, when the podnetworkconnectivitychecks
+// crd has been established, or send an error to the ready chan if an error has occurred.
+type timeToStartController struct {
+	factory.Controller
+	crdLister apiextensionslistersv1.CustomResourceDefinitionLister
+	C         chan error
+}
+
+func newTimeToStartController(crdInformer apiextensionsinformersv1.CustomResourceDefinitionInformer, recorder events.Recorder) *timeToStartController {
+	c := &timeToStartController{
+		crdLister: crdInformer.Lister(),
+		C:         make(chan error, 1),
+	}
+	c.Controller = factory.New().
+		WithInformers(crdInformer.Informer()).
+		WithSync(func(context.Context, factory.SyncContext) error {
+			established, err := podNetworkConnectivityCheckResourceDefinitionEstablished(c.crdLister)
+			if err != nil {
+				c.C <- err
+				return err
+			}
+			if established {
+				close(c.C)
+			}
+			return nil
+		}).
+		ToController("CheckEndpointsTimeToStart", recorder)
+	return c
+}
+
+func (c *timeToStartController) Ready() <-chan error {
+	return c.C
+}
+
+// stopController stops the process using os.Exit(1) if the podnetworkconnectivitychecks crd
+// stops being established after the process has started.
+type stopController struct {
+	factory.Controller
+	crdLister apiextensionslistersv1.CustomResourceDefinitionLister
+}
+
+func newStopController(crdInformer apiextensionsinformersv1.CustomResourceDefinitionInformer, recorder events.Recorder) *stopController {
+	c := &stopController{
+		crdLister: crdInformer.Lister(),
+	}
+	c.Controller = factory.New().
+		WithInformers(crdInformer.Informer()).
+		WithSync(func(context.Context, factory.SyncContext) error {
+			established, err := podNetworkConnectivityCheckResourceDefinitionEstablished(c.crdLister)
+			if err != nil {
+				return err
+			}
+			if !established {
```

Comment:
it can never go unestablished. Check for 404.

Reply:
Done.